### PR TITLE
[next] add tests for server actions

### PR DESCRIPTION
**Pull request by ztanner**
There aren't currently any tests for server actions which have special handling in the builder. This adds tests to help avoid regressions in future Next.js builder work.

### CODE REVIEW

File: .changeset/smooth-moles-lay.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+---
+---
```
Comment (ijjk) [inline .changeset/smooth-moles-lay.md]: I think we should be able to skip the changeset if we add the tests only label

File: packages/next/test/fixtures/00-app-dir-actions/app/client/static/[dynamic-static]/page.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+'use client';
+
+import { useState } from 'react';
+import { increment } from '../../../actions';
+
+export const dynamic = 'force-static';
```
Comment (ijjk) [inline packages/next/test/fixtures/00-app-dir-actions/app/client/static/[dynamic-static]/page.js]: This has no effect in `use client` is this being statically generated like we're hoping?

File: packages/next/test/fixtures/00-app-dir-actions/app/client/dynamic/[id]/page.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+'use client';
+
+import { useState } from 'react';
+import { increment } from '../../../actions';
+
+export const dynamic = 'force-dynamic';
```
Comment (ijjk) [inline packages/next/test/fixtures/00-app-dir-actions/app/client/dynamic/[id]/page.js]: Similar to https://github.com/vercel/vercel/pull/11482/files#r1577054712

File: packages/next/test/fixtures/00-app-dir-actions/app/rsc/static/[dynamic-static]/page.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { revalidatePath } from 'next/cache'
+
+export const dynamic = 'force-static'
```
Comment (ijjk) [inline packages/next/test/fixtures/00-app-dir-actions/app/rsc/static/[dynamic-static]/page.js]: Should we add `generateStaticParams` and test for some pre-generated paths as well as lazily generated?

File: .changeset/smooth-moles-lay.md (hunk 1 above)
Comment (ztanner) [reply .changeset/smooth-moles-lay.md]: Looks like this is required

**Review by ijjk — APPROVED** [review]
Looks good, thanks!

**Pull request merged**